redis/tests/test_cluster_async.rs

```diff
@@ -1457,6 +1457,42 @@ fn test_async_cluster_non_retryable_error_should_not_retry() {
     assert_eq!(completed.load(Ordering::SeqCst), 1);
 }
 
+#[test]
+fn test_async_cluster_can_be_created_with_partial_slot_coverage() {
+    let name = "test_async_cluster_can_be_created_with_partial_slot_coverage";
```

Comment:
Is it ever possible that slots would _overlap_? I've never encountered this but wondering if possible.

Reply:
It's not possible, according to the redis spec. Each slot can be managed only by one shard.

Reply:
Yeah, makes sense. I was wondering more in the scenario of a cluster that has briefly lost its mind, could it possibly serve out a list of nodes w/ overlapping slots? But I'm guessing not, and in any case I think the code here would probably handle it without issue.

Reply:
Putting aside the possibility of bugs in Redis code (see gif below), a single node won't return overlapping slots. Different nodes might have different views of the server, but that's a tomorrow-problem.

![image](https://github.com/redis-rs/redis-rs/assets/4604038/98408b7e-da98-4c1c-b11b-654ad31d7191)